diff --git a/advanced/Scripts/piholeDebug.sh b/advanced/Scripts/piholeDebug.sh
index 7c558127..c1f6a35c 100755
--- a/advanced/Scripts/piholeDebug.sh
+++ b/advanced/Scripts/piholeDebug.sh
@@ -520,7 +548,13 @@ disk_usage() {
 parse_locale() {
     local pihole_locale
     echo_current_diagnostic "Locale"
-    pihole_locale="$(locale)"
+    if [ -f "$(command -v locale)" ]; then
+        pihole_locale="$(locale)"
+    elif [ -f "$(command -v locale.new)" ]; then
+        pihole_locale="$(locale.new)"
+    else
+        pihole_locale="No locale command found"
+    fi
     parse_file "${pihole_locale}"
 }
 
