diff --git a/advanced/Scripts/piholeDebug.sh b/advanced/Scripts/piholeDebug.sh
index 7c55812..2150ba0 100755
--- a/advanced/Scripts/piholeDebug.sh
+++ b/advanced/Scripts/piholeDebug.sh
@@ -1277,6 +1277,14 @@ analyze_pihole_log() {
 }
 
 curl_to_tricorder() {
+    if [ -f /opt/etc/opkg.conf ]; then
+        log_write ""
+        log_write "   * ${COL_YELLOW}Upload is not supported in unofficial Entware package release!${COL_NC}"
+        log_write "   * You can manually upload the log when requested by an official Pi-hole developer: ${COL_CYAN}'cat /opt/var/log/pihole/pihole_debug.log | pihole tricorder'${COL_NC}"
+        log_write ""
+        return
+    fi
+
     # Users can submit their debug logs using curl (encrypted)
     log_write "    * Using ${COL_GREEN}curl${COL_NC} for transmission."
     # transmit the log via TLS and store the token returned in a variable
@@ -1339,7 +1339,7 @@ upload_to_tricorder() {
         # let the user know
         log_write "${INFO} Debug script running in automated mode"
         # and then decide again which tool to use to submit it
-        curl_to_tricorder
+        #curl_to_tricorder
         # If we're not running in automated mode,
     else
         echo ""
diff --git a/pihole b/pihole
index 7c84771..f4125a9 100755
--- a/pihole
+++ b/pihole
@@ -465,7 +469,12 @@ updateCheckFunc() {
 }
 
 unsupportedFunc(){
-  echo "Function not supported in Docker images"
+  if [ -n "${DOCKER_VERSION}" ]; then
+    echo "Function not supported in Docker images"
+  elif [ -f /opt/etc/opkg.conf ]; then
+    echo "Function not supported in unofficial Entware package release"
+  fi
+
   exit 0
 }
 
@@ -578,17 +578,17 @@ case "${1}" in
   "--allow-wild" | "allow-wild"   ) listFunc "$@";;
   "-d" | "debug"                  ) debugFunc "$@";;
   "-f" | "flush"                  ) flushFunc "$@";;
-  "-up" | "updatePihole"          ) updatePiholeFunc "$@";;
-  "-r"  | "reconfigure"           ) reconfigurePiholeFunc;;
+  "-up" | "updatePihole"          ) unsupportedFunc;;
+  "-r"  | "reconfigure"           ) unsupportedFunc;;
   "-g" | "updateGravity"          ) updateGravityFunc "$@";;
   "-l" | "logging"                ) piholeLogging "$@";;
-  "uninstall"                     ) uninstallFunc;;
+  "uninstall"                     ) unsupportedFunc;;
   "enable"                        ) piholeEnable true "$2";;
   "disable"                       ) piholeEnable false "$2";;
   "reloaddns"                     ) reloadDNS "reload";;
   "reloadlists"                   ) reloadDNS "reload-lists";;
   "setpassword"                   ) SetWebPassword "$@";;
-  "checkout"                      ) piholeCheckoutFunc "$@";;
+  "checkout"                      ) unsupportedFunc;;
   "updatechecker"                 ) shift; updateCheckFunc "$@";;
   "arpflush"                      ) arpFunc "$@";;
   "-t" | "tail"                   ) tailFunc "$2";;
