diff --git a/advanced/Scripts/piholeDebug.sh b/advanced/Scripts/piholeDebug.sh
index 7c558127..e8534db5 100755
--- a/advanced/Scripts/piholeDebug.sh
+++ b/advanced/Scripts/piholeDebug.sh
@@ -628,9 +628,16 @@ check_required_ports() {
     # Create an array for these ports in use
     ports_in_use=()
     # Sort the addresses and remove duplicates
-    while IFS= read -r line; do
-        ports_in_use+=( "$line" )
-    done < <( ss --listening --numeric --tcp --udp --processes --no-header )
+	if [ -f "$(command -v ss)" ]; then
+		while IFS= read -r line; do
+			ports_in_use+=( "$line" )
+		done < <( ss --listening --numeric --tcp --udp --processes --no-header )
+	else
+		local use_nestat=y
+		while IFS= read -r line; do
+			ports_in_use+=( "$line" )
+		done < <( netstat -tulnp | grep "tcp\|udp")
+	fi
 
     local ports_configured
     # Get all configured ports
@@ -645,11 +652,21 @@ check_required_ports() {
     for i in "${!ports_in_use[@]}"; do
         # loop through them and assign some local variables
         local service_name
-        service_name=$(echo "${ports_in_use[$i]}" | awk '{gsub(/users:\(\("/,"",$7);gsub(/".*/,"",$7);print $7}')
+		if [ -n "$use_nestat" ]; then
+			service_name="$(echo "${ports_in_use[$i]}" | awk '{print $7}' | cut -d '/' -f 2)"
+			[ -z "$service_name" ] && service_name="$(echo "${ports_in_use[$i]}" | awk '{print $6}' | cut -d '/' -f 2)"
+			[ "$service_name" = "-" ] && service_name=""
+		else
+			service_name=$(echo "${ports_in_use[$i]}" | awk '{gsub(/users:\(\("/,"",$7);gsub(/".*/,"",$7);print $7}')
+		fi
         local protocol_type
         protocol_type=$(echo "${ports_in_use[$i]}" | awk '{print $1}')
         local port_number
-        port_number="$(echo "${ports_in_use[$i]}" | awk '{print $5}')" #  | awk '{gsub(/^.*:/,"",$5);print $5}')
+		if [ -n "$use_nestat" ]; then
+			port_number="$(echo "${ports_in_use[$i]}" | awk '{print $4}')"
+		else
+			port_number="$(echo "${ports_in_use[$i]}" | awk '{print $5}')" #  | awk '{gsub(/^.*:/,"",$5);print $5}')
+		fi
 
         # Check if the right services are using the right ports
         if [[ ${ports_configured[*]} =~ $(echo "${port_number}" | rev | cut -d: -f1 | rev) ]]; then
diff --git a/pihole b/pihole
index 7c84771c..cd35f2fb 100755
--- a/pihole
+++ b/pihole
@@ -317,7 +317,12 @@ analyze_ports() {
   # function is getting called
   # Check individual address family/protocol combinations
   # For a healthy Pi-hole, they should all be up (nothing printed)
-  lv4="$(ss --ipv4 --listening --numeric --tcp --udp src :${port})"
+  if [ -f "$(command -v ss)" ]; then
+    lv4="$(ss --ipv4 --listening --numeric --tcp --udp src :${port})"
+  else
+    lv4="$(netstat -tuln -4 | grep ":${port} ")"
+  fi
+
   if grep -q "udp " <<< "${lv4}"; then
       echo -e "     ${TICK} UDP (IPv4)"
   else
@@ -328,7 +333,13 @@ analyze_ports() {
   else
       echo -e "     ${CROSS} TCP (IPv4)"
   fi
-  lv6="$(ss --ipv6 --listening --numeric --tcp --udp src :${port})"
+
+  if [ -f "$(command -v ss)" ]; then
+    lv6="$(ss --ipv6 --listening --numeric --tcp --udp src :${port})"
+  else
+    lv6="$(netstat -tuln -6 | grep ":${port} ")"
+  fi
+
   if grep -q "udp " <<< "${lv6}"; then
       echo -e "     ${TICK} UDP (IPv6)"
   else
