diff --git a/advanced/Scripts/updatecheck.sh b/advanced/Scripts/updatecheck.sh
index b325ee9c..2ab983c1 100755
--- a/advanced/Scripts/updatecheck.sh
+++ b/advanced/Scripts/updatecheck.sh
@@ -34,6 +34,14 @@ function get_remote_version() {
     fi
 }
 
+function get_remote_version_entware() {
+    if [[ "${1}" == "pi-hole" ]]; then
+        curl -s "https://api.github.com/repos/jacklul/entware-pi-hole/releases/latest" 2>/dev/null | jq --raw-output .tag_name || return 1
+    else
+        echo "null"
+    fi
+}
+
 function get_remote_hash() {
     git ls-remote "https://github.com/pi-hole/${1}" --tags "${2}" | awk '{print substr($0, 1,8);}' || return 1
 }
@@ -61,6 +69,12 @@ if [[ ! "${DOCKER_TAG}" =~ $regex ]]; then
     unset DOCKER_TAG
 fi
 
+# load current versions when running in Entware since we don't have locally cloned repos in this case
+if [ -f "${VERSION_FILE}" ]; then
+    # shellcheck disable=SC1090
+    . "${VERSION_FILE}"
+fi
+
 # used in cronjob
 if [[ "$1" == "reboot" ]]; then
     sleep 30
@@ -68,15 +82,6 @@ fi
 
 # get Core versions
 
-CORE_VERSION="$(get_local_version /opt/etc/.pihole)"
-addOrEditKeyValPair "${VERSION_FILE}" "CORE_VERSION" "${CORE_VERSION}"
-
-CORE_BRANCH="$(get_local_branch /opt/etc/.pihole)"
-addOrEditKeyValPair "${VERSION_FILE}" "CORE_BRANCH" "${CORE_BRANCH}"
-
-CORE_HASH="$(get_local_hash /opt/etc/.pihole)"
-addOrEditKeyValPair "${VERSION_FILE}" "CORE_HASH" "${CORE_HASH}"
-
 GITHUB_CORE_VERSION="$(get_remote_version pi-hole "${CORE_BRANCH}")"
 addOrEditKeyValPair "${VERSION_FILE}" "GITHUB_CORE_VERSION" "${GITHUB_CORE_VERSION}"
 
@@ -85,15 +90,6 @@ addOrEditKeyValPair "${VERSION_FILE}" "GITHUB_CORE_HASH" "${GITHUB_CORE_HASH}"
 
 # get Web versions
 
-WEB_VERSION="$(get_local_version "${ADMIN_INTERFACE_DIR}")"
-addOrEditKeyValPair "${VERSION_FILE}" "WEB_VERSION" "${WEB_VERSION}"
-
-WEB_BRANCH="$(get_local_branch "${ADMIN_INTERFACE_DIR}")"
-addOrEditKeyValPair "${VERSION_FILE}" "WEB_BRANCH" "${WEB_BRANCH}"
-
-WEB_HASH="$(get_local_hash "${ADMIN_INTERFACE_DIR}")"
-addOrEditKeyValPair "${VERSION_FILE}" "WEB_HASH" "${WEB_HASH}"
-
 GITHUB_WEB_VERSION="$(get_remote_version web "${WEB_BRANCH}")"
 addOrEditKeyValPair "${VERSION_FILE}" "GITHUB_WEB_VERSION" "${GITHUB_WEB_VERSION}"
 
@@ -102,15 +98,6 @@ addOrEditKeyValPair "${VERSION_FILE}" "GITHUB_WEB_HASH" "${GITHUB_WEB_HASH}"
 
 # get FTL versions
 
-FTL_VERSION="$(pihole-FTL version)"
-addOrEditKeyValPair "${VERSION_FILE}" "FTL_VERSION" "${FTL_VERSION}"
-
-FTL_BRANCH="$(pihole-FTL branch)"
-addOrEditKeyValPair "${VERSION_FILE}" "FTL_BRANCH" "${FTL_BRANCH}"
-
-FTL_HASH="$(pihole-FTL --hash)"
-addOrEditKeyValPair "${VERSION_FILE}" "FTL_HASH" "${FTL_HASH}"
-
 GITHUB_FTL_VERSION="$(get_remote_version FTL "${FTL_BRANCH}")"
 addOrEditKeyValPair "${VERSION_FILE}" "GITHUB_FTL_VERSION" "${GITHUB_FTL_VERSION}"
 
@@ -131,3 +118,21 @@ if [[ "${DOCKER_TAG}" ]]; then
     GITHUB_DOCKER_VERSION="$(get_remote_version docker-pi-hole "${docker_branch}")"
     addOrEditKeyValPair "${VERSION_FILE}" "GITHUB_DOCKER_VERSION" "${GITHUB_DOCKER_VERSION}"
 fi
+
+# get Entware package versions
+
+if [ -f /opt/etc/opkg.conf ]; then
+    ENTWARE_PACKAGE="$(opkg list-installed pi-hole* | awk '{print $1}' | head -n 1)"
+    addOrEditKeyValPair "${VERSION_FILE}" "ENTWARE_PACKAGE" "${ENTWARE_PACKAGE}"
+
+    ENTWARE_VERSION="$(opkg list-installed "${ENTWARE_PACKAGE}" | awk '{print $3}')"
+    addOrEditKeyValPair "${VERSION_FILE}" "ENTWARE_VERSION" "${ENTWARE_VERSION}"
+
+    # Check if the version is in the release version format
+    if [ "$ENTWARE_PACKAGE" = "pi-hole" ] && [[ ! "${ENTWARE_VERSION}" =~ $release_regex ]]; then
+        unset ENTWARE_PACKAGE
+    fi
+
+    GITHUB_ENTWARE_VERSION="$(get_remote_version_entware "$ENTWARE_PACKAGE")"
+    addOrEditKeyValPair "${VERSION_FILE}" "GITHUB_ENTWARE_VERSION" "${GITHUB_ENTWARE_VERSION}"
+fi
