diff --git a/advanced/Scripts/COL_TABLE b/advanced/Scripts/COL_TABLE
index 4344c7b8..defa0e56 100644
--- a/advanced/Scripts/COL_TABLE
+++ b/advanced/Scripts/COL_TABLE
@@ -1,5 +1,5 @@
 # Determine if terminal is capable of showing colors
-if ([ -t 1 ] && [ $(tput colors) -ge 8 ]) || [ "${WEBCALL}" ]; then
+if ([ -t 1 ] && [ $(tput colors) -ge 8 ]) 2> /dev/null || [ "${WEBCALL}" ]; then
   # Bold and underline may not show up on all clients
   # If something MUST be emphasized, use both
   COL_BOLD='[1m'
diff --git a/advanced/Scripts/piholeDebug.sh b/advanced/Scripts/piholeDebug.sh
index 7c558127..0bfbd015 100755
--- a/advanced/Scripts/piholeDebug.sh
+++ b/advanced/Scripts/piholeDebug.sh
@@ -1235,10 +1235,10 @@ spinner(){
         _start=$(date +%s)
 
         # Hide the cursor
-        tput civis > /dev/tty
+        tput civis > /dev/tty 2> /dev/null
 
         # ensures cursor is visible again, in case of premature exit
-        trap 'tput cnorm > /dev/tty' EXIT
+        trap 'tput cnorm > /dev/tty 2> /dev/null' EXIT
 
         while [ -d /proc/$_PID ]; do
             _elapsed=$(( $(date +%s) - _start ))
@@ -1255,7 +1255,7 @@ spinner(){
         printf "\r" >"$(tty)"
 
         # Restore cursor visibility
-        tput cnorm > /dev/tty
+        tput cnorm > /dev/tty 2> /dev/null
     fi
 }
 
