diff --git a/advanced/Scripts/piholeLogFlush.sh b/advanced/Scripts/piholeLogFlush.sh
index 4d97fec..9d12308 100755
--- a/advanced/Scripts/piholeLogFlush.sh
+++ b/advanced/Scripts/piholeLogFlush.sh
@@ -65,8 +65,13 @@ else
     # Delete most recent 24 hours from FTL's database, leave even older data intact (don't wipe out all history)
     deleted=$(pihole-FTL sqlite3 -ni "${DBFILE}" "DELETE FROM query_storage WHERE timestamp >= strftime('%s','now')-86400; select changes() from query_storage limit 1")
 
+    # Run with sudo only if we are not root already
+    if [[ ${EUID} -ne 0 ]]; then
+        SUDO="sudo"
+    fi
+
     # Restart pihole-FTL to force reloading history
-    sudo pihole restartdns
+    ${SUDO} pihole restartdns
 fi
 
 if [[ "$*" != *"quiet"* ]]; then
