diff --git a/advanced/Scripts/api.sh b/advanced/Scripts/api.sh
index 79fc90f4..6f08eeb2 100755
--- a/advanced/Scripts/api.sh
+++ b/advanced/Scripts/api.sh
@@ -10,6 +10,7 @@
 # This file is copyright under the latest version of the EUPL.
 # Please see LICENSE file for your rights under this license.
 
+source "/opt/share/pihole/utils.sh"
 
 # The basic usage steps are
 # 1) Test Availability of the API
@@ -26,7 +26,8 @@ TestAPIAvailability() {
     # Query the API URLs from FTL using CHAOS TXT local.api.ftl
     # The result is a space-separated enumeration of full URLs
     # e.g., "http://localhost:80/api/" "https://localhost:443/api/"
-    chaos_api_list="$(dig +short chaos txt local.api.ftl @127.0.0.1)"
+    #shellcheck disable=SC2046
+    chaos_api_list="$(dig +short chaos txt local.api.ftl @127.0.0.1 -p "$(getFTLConfigValue dns.port)")"
 
     # If the query was not successful, the variable is empty
     if [ -z "${chaos_api_list}" ]; then
