diff --git a/advanced/Scripts/piholeDebug.sh b/advanced/Scripts/piholeDebug.sh
index 7c55812..0116ad3 100755
--- a/advanced/Scripts/piholeDebug.sh
+++ b/advanced/Scripts/piholeDebug.sh
@@ -376,6 +376,20 @@ diagnose_operating_system() {
     # If DOCKER_VERSION is set (Sourced from /etc/pihole/versions at start of script), include this information in the debug output
     [ -n "${DOCKER_VERSION}" ] && log_write "${INFO} Pi-hole Docker Container: ${DOCKER_VERSION}"
 
+    # Existence of this file indicates Entware package is used
+    if [ -f /opt/etc/opkg.conf ];then
+        # shellcheck disable=SC2155
+        local entware_installed_package="$(opkg list-installed pi-hole)"
+        [ -z "${entware_installed_package}" ] && entware_installed_package="$(opkg list-installed pi-hole-dev)"
+        [ -n "${entware_installed_package}" ] && log_write "${INFO} Pi-hole Entware package: ${entware_installed_package}"
+
+        # Print informations about Entware release
+        if [ -f /opt/etc/entware-release ];then
+            log_write "${INFO} Entware release details:"
+            make_array_from_file "/opt/etc/entware-release"
+        fi
+    fi
+
     # If there is a /etc/*release file, it's probably a supported operating system, so we can
     if ls /etc/*release 1> /dev/null 2>&1; then
         # display the attributes to the user from the function made earlier
@@ -521,6 +535,9 @@ parse_locale() {
     local pihole_locale
     echo_current_diagnostic "Locale"
     pihole_locale="$(locale)"
+    if [ -f /opt/etc/opkg.conf ] && [ -f "$(command -v locale.new)" ]; then
+        pihole_locale="$(locale.new)"
+    fi
     parse_file "${pihole_locale}"
 }
 
