diff --git a/advanced/Scripts/piholeDebug.sh b/advanced/Scripts/piholeDebug.sh
index 7c558127..0b57bb6a 100755
--- a/advanced/Scripts/piholeDebug.sh
+++ b/advanced/Scripts/piholeDebug.sh
@@ -828,7 +828,7 @@ process_status(){
                 fi
             else
             # non-Docker system
-                if service "${i}" status | grep -E 'is\srunning' &> /dev/null; then
+                if /opt/etc/init.d/S55pihole-FTL check | grep 'alive' &> /dev/null; then
                     status_of_process="active"
                 else
                     status_of_process="inactive"
diff --git a/pihole b/pihole
index 7c84771c..2ddd1202 100755
--- a/pihole
+++ b/pihole
@@ -182,7 +182,7 @@ restartDNS() {
     fi
   else
     # A full restart has been requested
-    svc="service pihole-FTL restart"
+    svc="/opt/etc/init.d/S55pihole-FTL restart"
     str="Restarting DNS server"
     icon="${TICK}"
   fi
