diff --git a/src/gen_version.cmake b/src/gen_version.cmake
index a4130332..bf659c98 100644
--- a/src/gen_version.cmake
+++ b/src/gen_version.cmake
@@ -24,7 +24,7 @@ if(DEFINED ENV{GIT_HASH})
     set(GIT_HASH "$ENV{GIT_HASH}")
 else()
     execute_process(
-            COMMAND           git --no-pager describe --always --abbrev=8 --dirty
+            COMMAND           git --no-pager describe --always --abbrev=8
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
             OUTPUT_VARIABLE   GIT_HASH
             ERROR_QUIET
@@ -36,7 +36,7 @@ if(DEFINED ENV{GIT_VERSION})
     set(GIT_VERSION "$ENV{GIT_VERSION}")
 else()
     execute_process(
-            COMMAND           git --no-pager describe --tags --always --abbrev=8 --dirty
+            COMMAND           git --no-pager describe --tags --always --abbrev=8
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
             OUTPUT_VARIABLE   GIT_VERSION
             ERROR_QUIET
