diff --git a/src/zip/miniz/miniz.c b/src/zip/miniz/miniz.c
index 1195e620..237b9351 100644
--- a/src/zip/miniz/miniz.c
+++ b/src/zip/miniz/miniz.c
@@ -6230,6 +6230,8 @@ mz_bool mz_zip_writer_add_mem_ex_v2(mz_zip_archive *pZip, const char *pArchive_n
     level = level_and_flags & 0xF;
     store_data_uncompressed = ((!level) || (level_and_flags & MZ_ZIP_FLAG_COMPRESSED_DATA));
 
+    pArchive_name = strip_opt_from_path(pArchive_name);
+
     if ((!pZip) || (!pZip->m_pState) || (pZip->m_zip_mode != MZ_ZIP_MODE_WRITING) || ((buf_size) && (!pBuf)) || (!pArchive_name) || ((comment_size) && (!pComment)) || (level > MZ_UBER_COMPRESSION))
         return mz_zip_set_error(pZip, MZ_ZIP_INVALID_PARAMETER);
 
@@ -6891,6 +6893,8 @@ mz_bool mz_zip_writer_add_file(mz_zip_archive *pZip, const char *pArchive_name,
     uncomp_size = MZ_FTELL64(pSrc_file);
     MZ_FSEEK64(pSrc_file, 0, SEEK_SET);
 
+    pArchive_name = strip_opt_from_path(pArchive_name);
+
     status = mz_zip_writer_add_cfile(pZip, pArchive_name, pSrc_file, uncomp_size, pFile_time, pComment, comment_size, level_and_flags, NULL, 0, NULL, 0);
 
     MZ_FCLOSE(pSrc_file);
@@ -7476,6 +7480,8 @@ mz_bool mz_zip_add_mem_to_archive_file_in_place_v2(const char *pZip_filename, co
     if ((int)level_and_flags < 0)
         level_and_flags = MZ_DEFAULT_LEVEL;
 
+    pArchive_name = strip_opt_from_path(pArchive_name);
+
     if ((!pZip_filename) || (!pArchive_name) || ((buf_size) && (!pBuf)) || ((comment_size) && (!pComment)) || ((level_and_flags & 0xF) > MZ_UBER_COMPRESSION))
     {
         if (pErr)
@@ -7826,6 +7832,14 @@ mz_bool mz_zip_end(mz_zip_archive *pZip)
     return MZ_FALSE;
 }
 
+const char* strip_opt_from_path(const char* path)
+{
+    if (strncmp(path, "opt/", 4) == 0)
+        return path + 4;
+
+    return path;
+}
+
 #ifdef __cplusplus
 }
 #endif
diff --git a/src/zip/miniz/miniz.h b/src/zip/miniz/miniz.h
index d6a354bf..5fd3ec01 100644
--- a/src/zip/miniz/miniz.h
+++ b/src/zip/miniz/miniz.h
@@ -1415,6 +1415,8 @@ MINIZ_EXPORT void *mz_zip_extract_archive_file_to_heap_v2(const char *pZip_filen
 
 #endif /* #ifndef MINIZ_NO_ARCHIVE_WRITING_APIS */
 
+const char* strip_opt_from_path(const char* path);
+
 #ifdef __cplusplus
 }
 #endif
