diff --git a/src/api/info.c b/src/api/info.c
index 581fc0b..c0068bb 100644
--- a/src/api/info.c
+++ b/src/api/info.c
@@ -726,6 +726,7 @@ int api_info_version(struct ftl_conn *api)
 	cJSON *web_remote = JSON_NEW_OBJECT();
 	cJSON *ftl_remote = JSON_NEW_OBJECT();
 	cJSON *docker = JSON_NEW_OBJECT();
+	cJSON *entware = JSON_NEW_OBJECT();
 
 	FILE *fp = fopen(VERSIONS_FILE, "r");
 	if(!fp)
@@ -791,6 +792,10 @@ int api_info_version(struct ftl_conn *api)
 			JSON_COPY_STR_TO_OBJECT(docker, "local", value);
 		else if(strcmp(key, "GITHUB_DOCKER_VERSION") == 0)
 			JSON_COPY_STR_TO_OBJECT(docker, "remote", value);
+		else if(strcmp(key, "ENTWARE_VERSION") == 0)
+			JSON_COPY_STR_TO_OBJECT(entware, "local", value);
+		else if(strcmp(key, "GITHUB_ENTWARE_VERSION") == 0)
+			JSON_COPY_STR_TO_OBJECT(entware, "remote", value);
 	}
 
 	// Free allocated memory and release file pointer
@@ -839,6 +844,11 @@ int api_info_version(struct ftl_conn *api)
 	JSON_ADD_NULL_IF_NOT_EXISTS(docker, "remote");
 	JSON_ADD_ITEM_TO_OBJECT(version, "docker", docker);
 
+	// Add nulls to entware if we didn't find any version
+	JSON_ADD_NULL_IF_NOT_EXISTS(entware, "local");
+	JSON_ADD_NULL_IF_NOT_EXISTS(entware, "remote");
+	JSON_ADD_ITEM_TO_OBJECT(version, "entware", entware);
+
 	return 0;
 }
 
