diff --git a/scripts/pi-hole/js/footer.js b/scripts/pi-hole/js/footer.js
index 65726b0b..6fdd8fa3 100644
--- a/scripts/pi-hole/js/footer.js
+++ b/scripts/pi-hole/js/footer.js
@@ -580,14 +580,14 @@ function updateVersionInfo() {
       }
     });
 
-    if (dockerUpdate)
+    /*if (dockerUpdate)
       $("#update-hint").html(
         'To install updates, <a href="https://github.com/pi-hole/docker-pi-hole#upgrading-persistence-and-customizations" rel="noopener" target="_blank">replace this old container with a fresh upgraded image</a>.'
       );
     else if (updateAvailable)
       $("#update-hint").html(
         'To install updates, run <code><a href="https://docs.pi-hole.net/main/update/" rel="noopener" target="_blank">pihole -up</a></code>.'
-      );
+      );*/
 
     clearTimeout(versionTimer);
     versionTimer = utils.setTimer(updateVersionInfo, REFRESH_INTERVAL.version);
