diff --git a/scripts/js/footer.js b/scripts/js/footer.js
index 65726b0b..1265ff21 100644
--- a/scripts/js/footer.js
+++ b/scripts/js/footer.js
@@ -485,6 +485,8 @@ function updateVersionInfo() {
     var version = data.version;
     var updateAvailable = false;
     var dockerUpdate = false;
+    var entwareUpdate = false;
+    var packaged = version.docker.local || version.entware.local;
     var isDocker = false;
     $("#versions").empty();
 
@@ -496,6 +498,14 @@ function updateVersionInfo() {
         hash: null,
         url: "https://github.com/pi-hole/docker-pi-hole/releases",
       },
+      {
+        name: "Entware package",
+        local: version.entware.local,
+        remote: version.entware.remote,
+        branch: null,
+        hash: null,
+        url: "https://github.com/jacklul/entware-pi-hole/releases",
+      },
       {
         name: "Core",
         local: version.core.local.version,
@@ -562,6 +572,11 @@ function updateVersionInfo() {
           dockerUpdate = true;
         }
 
+        if (v.name === "Entware package" && versionCompare(v.local, v.remote) === -1) {
+          updateComponentAvailable = true;
+          entwareUpdate = true;
+        }
+
         // Display update information of individual components only if we are not running in a Docker container
         if ((!isDocker || (isDocker && v.name === "Docker Tag")) && updateComponentAvailable) {
           $("#versions").append(
@@ -584,7 +599,11 @@ function updateVersionInfo() {
       $("#update-hint").html(
         'To install updates, <a href="https://github.com/pi-hole/docker-pi-hole#upgrading-persistence-and-customizations" rel="noopener" target="_blank">replace this old container with a fresh upgraded image</a>.'
       );
-    else if (updateAvailable)
+    else if (entwareUpdate)
+      $("#update-hint").html(
+        'To install updates, run <code><a href="https://github.com/jacklul/entware-pi-hole/" rel="noopener" target="_blank">opkg update && opkg upgrade pi-hole</a></code>.'
+      );
+    else if (updateAvailable && !packaged)
       $("#update-hint").html(
         'To install updates, run <code><a href="https://docs.pi-hole.net/main/update/" rel="noopener" target="_blank">pihole -up</a></code>.'
       );
