diff --git a/advanced/Scripts/piholeDebug.sh b/advanced/Scripts/piholeDebug.sh
index 741ff2f4..1e9513b3 100755
--- a/advanced/Scripts/piholeDebug.sh
+++ b/advanced/Scripts/piholeDebug.sh
@@ -769,6 +769,12 @@ process_status(){
         if command -v systemctl &> /dev/null; then
             # get its status via systemctl
             status_of_process=$(systemctl is-active "${i}")
+        elif [ -f /opt/etc/opkg.conf ]; then
+            if /opt/etc/init.d/S65pihole-FTL check | grep -q 'alive' &> /dev/null; then
+                status_of_process="active"
+            else
+                status_of_process="inactive"
+            fi
         else
             # Otherwise, use the service command and mock the output of `systemctl is-active`
 
diff --git a/automated install/basic-install.sh b/automated install/basic-install.sh
index 6cc69008..c103d528 100755
--- a/automated install/basic-install.sh	
+++ b/automated install/basic-install.sh	
@@ -1236,6 +1236,8 @@ stop_service() {
     printf "  %b %s..." "${INFO}" "${str}"
     if is_command systemctl; then
         systemctl -q stop "${1}" || true
+    elif [ -f /opt/etc/opkg.conf ]; then
+        /opt/etc/init.d/S65pihole-FTL stop >/dev/null || true
     else
         service "${1}" stop >/dev/null || true
     fi
@@ -1252,6 +1254,8 @@ restart_service() {
     if is_command systemctl; then
         # use that to restart the service
         systemctl -q restart "${1}"
+    elif [ -f /opt/etc/opkg.conf ]; then
+        /opt/etc/init.d/S65pihole-FTL restart >/dev/null || true
     else
         # Otherwise, fall back to the service command
         service "${1}" restart >/dev/null
diff --git a/advanced/Scripts/piholeLogFlush.sh b/advanced/Scripts/piholeLogFlush.sh
index 34d96318..f905f045 100755
--- a/advanced/Scripts/piholeLogFlush.sh
+++ b/advanced/Scripts/piholeLogFlush.sh
@@ -108,13 +108,14 @@ else
     fi
 
     # Stop FTL to make sure it doesn't write to the database while we're deleting data
-    service pihole-FTL stop
+    /opt/etc/init.d/S65pihole-FTL stop
 
     # Delete most recent 24 hours from FTL's database, leave even older data intact (don't wipe out all history)
     deleted=$(pihole-FTL sqlite3 -ni "${DBFILE}" "DELETE FROM query_storage WHERE timestamp >= strftime('%s','now')-86400; select changes() from query_storage limit 1")
 
     # Restart FTL
-    service pihole-FTL restart
+    /opt/etc/init.d/S65pihole-FTL restart
+
     if [[ "$*" != *"quiet"* ]]; then
         echo -e "${OVER}  ${TICK} Deleted ${deleted} queries from long-term query database"
     fi
