diff --git a/advanced/Scripts/piholeDebug.sh b/advanced/Scripts/piholeDebug.sh
index 7c55812..05eb93d 100755
--- a/advanced/Scripts/piholeDebug.sh
+++ b/advanced/Scripts/piholeDebug.sh
@@ -1296,7 +1296,12 @@ upload_to_tricorder() {
     local username="pihole"
     # Set the permissions and owner
     chmod 640 ${PIHOLE_DEBUG_LOG}
-    chown "$USER":"${username}" ${PIHOLE_DEBUG_LOG}
+
+    if [ -n "$(id -u pihole 2> /dev/null)" ]; then
+        chown "$USER":"${username}" ${PIHOLE_DEBUG_LOG}
+    else
+        chown "$USER":"$USER" ${PIHOLE_DEBUG_LOG}
+    fi
 
     # Let the user know debugging is complete with something strikingly visual
     log_write ""
diff --git a/advanced/Templates/pihole-FTL-prestart.sh b/advanced/Templates/pihole-FTL-prestart.sh
index c681782..1443a90 100755
--- a/advanced/Templates/pihole-FTL-prestart.sh
+++ b/advanced/Templates/pihole-FTL-prestart.sh
@@ -9,16 +9,19 @@ utilsfile="${PI_HOLE_SCRIPT_DIR}/utils.sh"
 # Get file paths
 FTL_PID_FILE="$(getFTLConfigValue files.pid)"
 
+# In Entware user pihole might not exist
+[ -f /opt/etc/opkg.conf ] && [ -z "$(id -u pihole 2> /dev/null)" ] && NO_PIHOLE_USER=y
+
 # Ensure that permissions are set so that pihole-FTL can edit all necessary files
 # shellcheck disable=SC2174
 mkdir -pm 0640 /opt/var/log/pihole
-chown -R pihole:pihole /opt/etc/pihole /opt/var/log/pihole
+[ -z "$NO_PIHOLE_USER" ] && chown -R pihole:pihole /opt/etc/pihole /opt/var/log/pihole
 chmod -R 0640 /opt/var/log/pihole
 chmod -R 0660 /opt/etc/pihole
 
 # Logrotate config file need to be owned by root and must not be writable by group and others
-chown root:root /opt/etc/pihole/logrotate
+chown 0:0 /opt/etc/pihole/logrotate
 chmod 0644 /opt/etc/pihole/logrotate
 
 # allow all users to enter the directories
 chmod 0755 /opt/etc/pihole /opt/var/log/pihole
@@ -28,7 +28,14 @@ chmod 0755 /etc/pihole /var/log/pihole
 find /opt/etc/pihole/ -type d -exec chmod 0755 {} \;
 
 # Touch files to ensure they exist (create if non-existing, preserve if existing)
-[ -f "${FTL_PID_FILE}" ] || install -D -m 644 -o pihole -g pihole /dev/null "${FTL_PID_FILE}"
-[ -f /opt/var/log/pihole/FTL.log ] || install -m 640 -o pihole -g pihole /dev/null /opt/var/log/pihole/FTL.log
-[ -f /opt/var/log/pihole/pihole.log ] || install -m 640 -o pihole -g pihole /dev/null /opt/var/log/pihole/pihole.log
-[ -f /opt/etc/pihole/dhcp.leases ] || install -m 644 -o pihole -g pihole /dev/null /opt/etc/pihole/dhcp.leases
+if [ -z "$NO_PIHOLE_USER" ]; then
+    [ -f "${FTL_PID_FILE}" ] || install -D -m 644 -o pihole -g pihole /dev/null "${FTL_PID_FILE}"
+    [ -f /opt/var/log/pihole/FTL.log ] || install -m 640 -o pihole -g pihole /dev/null /opt/var/log/pihole/FTL.log
+    [ -f /opt/var/log/pihole/pihole.log ] || install -m 640 -o pihole -g pihole /dev/null /opt/var/log/pihole/pihole.log
+    [ -f /opt/etc/pihole/dhcp.leases ] || install -m 644 -o pihole -g pihole /dev/null /opt/etc/pihole/dhcp.leases
+else
+    [ -f "${FTL_PID_FILE}" ] || install -D -m 644 /dev/null "${FTL_PID_FILE}"
+    [ -f /opt/var/log/pihole/FTL.log ] || install -m 640 /dev/null /opt/var/log/pihole/FTL.log
+    [ -f /opt/var/log/pihole/pihole.log ] || install -m 640 /dev/null /opt/var/log/pihole/pihole.log
+    [ -f /opt/etc/pihole/dhcp.leases ] || install -m 644 /dev/null /opt/etc/pihole/dhcp.leases
+fi

diff --git a/gravity.sh b/gravity.sh
index 522d92f..bb8f35b 100755
--- a/gravity.sh
+++ b/gravity.sh
@@ -59,11 +59,14 @@ gravityTEMPfile="${GRAVITYDB}_temp"
 gravityBCKdir="${gravityDIR}/gravity_backups"
 gravityBCKfile="${gravityBCKdir}/gravity.db"
 
+# In Entware user pihole might not exist
+[ -f /opt/etc/opkg.conf ] && [ -z "$(id -u pihole 2> /dev/null)" ] && NO_PIHOLE_USER=y
+
 fix_owner_permissions() {
   # Fix ownership and permissions for the specified file
   # User and group are set to pihole:pihole
   # Permissions are set to 664 (rw-rw-r--)
-  chown pihole:pihole "${1}"
+  [ -z "$NO_PIHOLE_USER" ] && chown pihole:pihole "${1}"
   chmod 664 "${1}"
 
   # Ensure the containing directory is group writable
