diff --git a/advanced/Scripts/updatecheck.sh b/advanced/Scripts/updatecheck.sh
index 23abf7a6..904920be 100755
--- a/advanced/Scripts/updatecheck.sh
+++ b/advanced/Scripts/updatecheck.sh
@@ -9,20 +9,36 @@
 set -o pipefail
 
 function get_local_branch() {
-    # Return active branch
-    cd "${1}" 2>/dev/null || { echo "null"; return; }
-    git rev-parse --abbrev-ref HEAD || echo "null"
+    case "$1" in
+        "/opt/etc/.pihole")
+            sed -n 's/^CORE_BRANCH=//p' /opt/share/pihole/versions || echo "null"
+        ;;
+        "${ADMIN_INTERFACE_DIR}")
+            sed -n 's/^WEB_BRANCH=//p' /opt/share/pihole/versions || echo "null"
+        ;;
+    esac
 }
 
 function get_local_version() {
-    # Return active version
-    cd "${1}" 2>/dev/null || { echo "null"; return; }
-    git describe --tags --always 2>/dev/null || echo "null"
+    case "$1" in
+        "/opt/etc/.pihole")
+            sed -n 's/^CORE_VERSION=//p' /opt/share/pihole/versions || echo "null"
+        ;;
+        "${ADMIN_INTERFACE_DIR}")
+            sed -n 's/^WEB_VERSION=//p' /opt/share/pihole/versions || echo "null"
+        ;;
+    esac
 }
 
 function get_local_hash() {
-    cd "${1}" 2>/dev/null || { echo "null"; return; }
-    git rev-parse --short=8 HEAD || echo "null"
+    case "$1" in
+        "/opt/etc/.pihole")
+            sed -n 's/^CORE_HASH=//p' /opt/share/pihole/versions || echo "null"
+        ;;
+        "${ADMIN_INTERFACE_DIR}")
+            sed -n 's/^WEB_HASH=//p' /opt/share/pihole/versions || echo "null"
+        ;;
+    esac
 }
 
 function get_remote_version() {
@@ -34,6 +59,14 @@ function get_remote_version() {
     fi
 }
 
+function get_remote_version_entware() {
+    if [[ "${1}" == "pi-hole" ]]; then
+        curl -s "https://api.github.com/repos/jacklul/entware-pi-hole/releases/latest" 2>/dev/null | jq --raw-output .tag_name || return 1
+    else
+        echo "null"
+    fi
+}
+
 function get_remote_hash() {
     git ls-remote "https://github.com/pi-hole/${1}" --tags "${2}" | awk '{print substr($0, 1,8);}' || echo "null"
 }
@@ -135,3 +168,18 @@ if [[ "${DOCKER_TAG}" ]]; then
     GITHUB_DOCKER_VERSION="$(get_remote_version docker-pi-hole "${docker_branch}")"
     addOrEditKeyValPair "${VERSION_FILE}" "GITHUB_DOCKER_VERSION" "${GITHUB_DOCKER_VERSION}"
 fi
+
+if [ -f /opt/etc/opkg.conf ]; then
+    ENTWARE_PACKAGE="$(opkg list-installed pi-hole* | awk '{print $1}' | head -n 1)"
+    addOrEditKeyValPair "${VERSION_FILE}" "ENTWARE_PACKAGE" "${ENTWARE_PACKAGE}"
+
+    ENTWARE_VERSION="$(opkg list-installed "${ENTWARE_PACKAGE}" | awk '{print $3}')"
+    addOrEditKeyValPair "${VERSION_FILE}" "ENTWARE_VERSION" "${ENTWARE_VERSION}"
+
+    if [ "$ENTWARE_PACKAGE" = "pi-hole" ] && [[ ! "${ENTWARE_VERSION}" =~ $release_regex ]]; then
+        unset ENTWARE_PACKAGE
+    fi
+
+    GITHUB_ENTWARE_VERSION="$(get_remote_version_entware "$ENTWARE_PACKAGE")"
+    addOrEditKeyValPair "${VERSION_FILE}" "GITHUB_ENTWARE_VERSION" "${GITHUB_ENTWARE_VERSION}"
+fi
