diff --git a/advanced/Scripts/piholeDebug.sh b/advanced/Scripts/piholeDebug.sh
index 741ff2f4..1e9513b3 100755
--- a/advanced/Scripts/piholeDebug.sh
+++ b/advanced/Scripts/piholeDebug.sh
@@ -766,8 +766,14 @@ process_status(){
     for i in "${PIHOLE_PROCESSES[@]}"; do
         local status_of_process
 
+        if [ -f /opt/etc/opkg.conf ]; then
+            status_of_process="inactive"
+
+            if [ "${i}" == "pihole-FTL" ] && /opt/etc/init.d/S65pihole-FTL check | grep -q 'alive' &> /dev/null; then
+                status_of_process="active"
+            fi
         # If systemd
-        if command -v systemctl &> /dev/null; then
+        elif command -v systemctl &> /dev/null; then
             # get its status via systemctl
             status_of_process=$(systemctl is-active "${i}")
         else
diff --git a/advanced/Scripts/piholeLogFlush.sh b/advanced/Scripts/piholeLogFlush.sh
index 34d96318..f905f045 100755
--- a/advanced/Scripts/piholeLogFlush.sh
+++ b/advanced/Scripts/piholeLogFlush.sh
@@ -108,13 +108,14 @@ else
     fi
 
     # Stop FTL to make sure it doesn't write to the database while we're deleting data
-    service pihole-FTL stop
+    /opt/etc/init.d/S65pihole-FTL stop
 
     # Delete most recent 24 hours from FTL's database, leave even older data intact (don't wipe out all history)
     deleted=$(pihole-FTL sqlite3 -ni "${DBFILE}" "DELETE FROM query_storage WHERE timestamp >= strftime('%s','now')-86400; select changes() from query_storage limit 1")
 
     # Restart FTL
-    service pihole-FTL restart
+    /opt/etc/init.d/S65pihole-FTL restart
+
     if [[ "$*" != *"quiet"* ]]; then
         echo -e "${OVER}  ${TICK} Deleted ${deleted} queries from long-term query database"
     fi
diff --git a/automated install/basic-install.sh b/automated install/basic-install.sh
index 6cc69008..c103d528 100755
--- a/automated install/basic-install.sh	
+++ b/automated install/basic-install.sh	
@@ -1232,7 +1232,14 @@ stop_service() {
     # Can softfail, as process may not be installed when this is called
     local str="Stopping ${1} service"
     printf "  %b %s..." "${INFO}" "${str}"
-    if is_command systemctl; then
+    if [ -f /opt/etc/opkg.conf ]; then
+        if [ "${1}" == "pihole-FTL" ]; then
+            /opt/etc/init.d/S65pihole-FTL stop &>/dev/null || true
+        else
+            printf "%b  %b %s...\\n" "${OVER}" "${CROSS}" "${str}"
+            return
+        fi
+    elif is_command systemctl; then
         systemctl -q stop "${1}" || true
     else
         service "${1}" stop >/dev/null || true
@@ -1245,8 +1252,15 @@ restart_service() {
     # Local, named variables
     local str="Restarting ${1} service"
     printf "  %b %s..." "${INFO}" "${str}"
+    if [ -f /opt/etc/opkg.conf ]; then
+        if [ "${1}" == "pihole-FTL" ]; then
+            /opt/etc/init.d/S65pihole-FTL restart &>/dev/null || true
+        else
+            printf "%b  %b %s...\\n" "${OVER}" "${CROSS}" "${str}"
+            return
+        fi
     # If systemctl exists,
-    if is_command systemctl; then
+    elif is_command systemctl; then
         # use that to restart the service
         systemctl -q restart "${1}"
     else
@@ -1261,8 +1275,11 @@ enable_service() {
     # Local, named variables
     local str="Enabling ${1} service to start on reboot"
     printf "  %b %s..." "${INFO}" "${str}"
+    if [ -f /opt/etc/opkg.conf ]; then
+        printf "%b  %b %s...\\n" "${OVER}" "${CROSS}" "${str}"
+        return
     # If systemctl exists,
-    if is_command systemctl; then
+    elif is_command systemctl; then
         # use that to enable the service
         systemctl -q enable "${1}"
     else
@@ -1277,8 +1294,11 @@ disable_service() {
     # Local, named variables
     local str="Disabling ${1} service"
     printf "  %b %s..." "${INFO}" "${str}"
+    if [ -f /opt/etc/opkg.conf ]; then
+        printf "%b  %b %s...\\n" "${OVER}" "${CROSS}" "${str}"
+        return
     # If systemctl exists,
-    if is_command systemctl; then
+    elif is_command systemctl; then
         # use that to disable the service
         systemctl -q disable "${1}"
     else
@@ -1289,8 +1309,14 @@ disable_service() {
 }
 
 check_service_active() {
+    if [ -f /opt/etc/opkg.conf ]; then
+        if [ "${1}" == "pihole-FTL" ]; then
+            /opt/etc/init.d/S65pihole-FTL check | grep -q 'alive' &> /dev/null
+        else
+            return 1
+        fi
     # If systemctl exists,
-    if is_command systemctl; then
+    elif is_command systemctl; then
         # use that to check the status of the service
         systemctl -q is-enabled "${1}" 2>/dev/null
     else
