diff --git a/test/run.sh b/test/run.sh
index 5d490a82..dfd08dac 100755
--- a/test/run.sh
+++ b/test/run.sh
@@ -83,6 +83,33 @@ export FTLCONF_files_pcap='*123#./test/pcap'
 # Prepare gdb session
 echo "handle SIGHUP nostop SIGPIPE nostop SIGTERM nostop SIG32 nostop SIG33 nostop SIG34 nostop SIG35 nostop SIG41 nostop" > /root/.gdbinit
 
+declare -A LINKS=(
+    ["/etc/pihole"]="/opt/etc/pihole"
+    ["/var/log/pihole"]="/opt/var/log/pihole"
+    ["/var/www/html"]="/opt/share/pihole/www"
+    ["/run"]="/opt/var/run"
+    ["/tmp"]="/opt/tmp"
+    ["/etc/resolv.conf"]="/opt/etc/resolv.conf"
+    ["/etc/hosts"]="/opt/etc/hosts"
+    ["/etc/ethers"]="/opt/etc/ethers"
+    ["/etc/dnsmasq.d"]="/opt/etc/dnsmasq.d"
+    ["/var/lib/misc/dnsmasq.leases"]="/opt/var/lib/misc/dnsmasq.leases"
+    ["/etc/dnsmasq.conf"]="/opt/etc/dnsmasq.conf"
+    ["/var/run/dnsmasq.pid"]="/opt/var/run/dnsmasq.pid"
+)
+
+for KEY in "${!LINKS[@]}"; do
+    DIR="$(dirname "${LINKS[$KEY]}")"
+
+    if [ ! -d "$DIR" ]; then
+        mkdir -pvm 0777 "$DIR"
+    fi
+
+    if [ -e "$KEY" ] && [ ! -L "${LINKS[$KEY]}" ]; then
+        ln -sv "$KEY" "${LINKS[$KEY]}"
+    fi
+done
+
 # Start FTL
 if ! su pihole -s /bin/sh -c /home/pihole/pihole-FTL; then
   echo "pihole-FTL failed to start"
@@ -119,10 +119,10 @@ $BATS -p "test/test_suite.bats"
 RET=$?
 
 curl_to_tricorder() {
-  curl --silent --upload-file "${1}" https://tricorder.pi-hole.net
+  echo "null"
 }
 
-if [[ $RET != 0 ]]; then
+if false && [[ $RET != 0 ]]; then
   echo -n "pihole/pihole.log: "
   curl_to_tricorder /var/log/pihole/pihole.log
   echo ""
diff --git a/test/pihole.toml b/test/pihole.toml
index 0b433266..85d8a45c 100644
--- a/test/pihole.toml
+++ b/test/pihole.toml
@@ -665,29 +665,29 @@
     #
     # Allowed values are:
     #     A valid TLS certificate file (*.pem)
-    cert = "/etc/pihole/test.pem" ### CHANGED, default = "/etc/pihole/tls.pem"
+    cert = "/etc/pihole/test.pem" ### CHANGED, default = "/opt/etc/pihole/tls.pem"
 
     # Number of days the automatically generated self-signed TLS/SSL certificate will be
     # valid for.
     #
     # Defaults to 47 days. A minimum of 7 days is enforced.
     # Some devices may enforce shorter validity ranges. Note that defining a lower
     # validity range may require you to accept the self-signed certificate more often in
     # your browser.
     # Pi-hole will regenerate certificates it created itself two days prior to expiration.
     # If you are using your own certificate, you need to regenerate it yourself. In this
     # case, it is advised to set the validity range to 0 days, so that Pi-hole does not
     # try to regenerate your certificate. If you set the validity range to 0 days and
     # still try to generate a certificate, Pi-hole will set a fixed validity range of
     # roughly 30 years for the certificate.
     validity = 47

   [webserver.paths]
     # Server root on the host
     #
     # Allowed values are:
     #     A valid path
-    webroot = "/var/www/html"
+    webroot = "/opt/share/pihole/www"
 
     # Sub-directory of the root containing the web interface
     #
@@ -822,32 +822,32 @@
   #
   # Allowed values are:
   #     Any writable file
-  pid = "/run/pihole-FTL.pid"
+  pid = "/opt/var/run/pihole-FTL.pid"
 
   # The location of FTL's long-term database
   #
   # Allowed values are:
   #     Any FTL database
-  database = "/etc/pihole/pihole-FTL.db"
+  database = "/opt/etc/pihole/pihole-FTL.db"
 
   # The location of Pi-hole's gravity database
   #
   # Allowed values are:
   #     Any Pi-hole gravity database
-  gravity = "/etc/pihole/gravity.db"
+  gravity = "/opt/etc/pihole/gravity.db"
 
   # A temporary directory where Pi-hole can store files during gravity updates. This
   # directory must be writable by the user running gravity (typically pihole).
   #
   # Allowed values are:
   #     Any existing world-writable writable directory
-  gravity_tmp = "/tmp"
+  gravity_tmp = "/opt/tmp"
 
   # The database containing MAC -> Vendor information for the network table
   #
   # Allowed values are:
   #     Any Pi-hole macvendor database
-  macvendor = "/etc/pihole/macvendor.db"
+  macvendor = "/opt/etc/pihole/macvendor.db"
 
   # An optional file containing a pcap capture of the network traffic. This file is used
   # for debugging purposes only. If you don't know what this is, you don't need it.
@@ -871,19 +871,19 @@
     #
     # Allowed values are:
     #     any writable file
-    ftl = "/var/log/pihole/FTL.log"
+    ftl = "/opt/var/log/pihole/FTL.log"
 
     # The log file used by the embedded dnsmasq DNS server
     #
     # Allowed values are:
     #     Any writable file
-    dnsmasq = "/var/log/pihole/pihole.log"
+    dnsmasq = "/opt/var/log/pihole/pihole.log"
 
     # The log file used by the webserver
     #
     # Allowed values are:
     #     Any writable file
-    webserver = "/var/log/pihole/webserver.log"
+    webserver = "/opt/var/log/pihole/webserver.log"
 
 [misc]
   # Using privacy levels you can specify which level of detail you want to see in your
@@ -1305,7 +1305,7 @@
   # Warning: This is an advanced setting and should only be used with care.
   # Incorrectly formatted or config files specifying options which can only be defined
   # once can result in conflicts with the automatic configuration of Pi-hole (see
-  # /etc/pihole/dnsmasq.conf) and may stop DNS resolution from working.
+  # /opt/etc/pihole/dnsmasq.conf) and may stop DNS resolution from working.
   #
   # Allowed values are:
   #     true or false
diff --git a/test/test_suite.bats b/test/test_suite.bats
index 1891abde..52aa6f3d 100644
--- a/test/test_suite.bats
+++ b/test/test_suite.bats
@@ -1755,15 +1755,15 @@
   printf "%s\n" "${lines[@]}"
   [[ ${lines[0]} == 'Hello, world 1!' ]]
   [[ ${lines[1]} == 'Hello, world 2!' ]]
-  [[ ${lines[2]} == '[string "/var/www/html/broken_lua_2.lp"]:4: Cannot include [/var/www/html/does_not_exist.lp]: not found' ]]
+  [[ ${lines[2]} == '[string "/opt/share/pihole/www/broken_lua_2.lp"]:4: Cannot include [/opt/share/pihole/www/does_not_exist.lp]: not found' ]]
   [[ ${lines[3]} == 'stack traceback:' ]]
   [[ ${lines[4]} == "	[C]: in field 'include'" ]]
-  [[ ${lines[5]} == '	[string "/var/www/html/broken_lua.lp"]:4: in main chunk' ]]
+  [[ ${lines[5]} == '	[string "/opt/share/pihole/www/broken_lua.lp"]:4: in main chunk' ]]
   [[ ${lines[6]} == 'aborting' ]]
   [[ ${lines[7]} == '' ]]
 
   # Check if the error is logged (-F = fixed string (no regex), -q = quiet)
-  run grep -qF 'LSP Kepler: call failed: runtime error: [string "/var/www/html/broken_lua_2.lp"]:4: Cannot include [/var/www/html/does_not_exist.lp]: not found' /var/log/pihole/webserver.log
+  run grep -qF 'LSP Kepler: call failed: runtime error: [string "/opt/share/pihole/www/broken_lua_2.lp"]:4: Cannot include [/opt/share/pihole/www/does_not_exist.lp]: not found' /opt/var/log/pihole/webserver.log
   [[ $status == 0 ]]
 }
 
@@ -1522,7 +1522,7 @@
 
   run bash -c './pihole-FTL --config dns.revServers "[\"true,1.1.1.1,def,ghi\"]"'
   printf "%s\n" "${lines[@]}"
-  [[ ${lines[0]} == 'New dnsmasq configuration is not valid ('*'resolve at line '*' of /etc/pihole/dnsmasq.conf.temp: "rev-server=1.1.1.1,def"), config remains unchanged' ]]
+  [[ ${lines[0]} == 'New dnsmasq configuration is not valid ('*'resolve at line '*' of /opt/etc/pihole/dnsmasq.conf.temp: "rev-server=1.1.1.1,def"), config remains unchanged' ]]
   [[ $status == 3 ]]
 
   run bash -c './pihole-FTL --config webserver.api.excludeClients "[\".*\",\"$$$\",\"[[[\"]"'
@@ -1774,13 +1774,13 @@
 }
 
 @test "Expected number of config file rotations" {
-  run bash -c 'grep -c "INFO: Config file written to /etc/pihole/pihole.toml" /var/log/pihole/FTL.log'
+  run bash -c 'grep -c "INFO: Config file written to /opt/etc/pihole/pihole.toml" /var/log/pihole/FTL.log'
   printf "%s\n" "${lines[@]}"
   [[ ${lines[0]} == "3" ]]
-  run bash -c 'grep -c "DEBUG_CONFIG: Config file written to /etc/pihole/dnsmasq.conf" /var/log/pihole/FTL.log'
+  run bash -c 'grep -c "DEBUG_CONFIG: Config file written to /opt/etc/pihole/dnsmasq.conf" /var/log/pihole/FTL.log'
   printf "%s\n" "${lines[@]}"
   [[ ${lines[0]} == "1" ]]
-  run bash -c 'grep -c "DEBUG_CONFIG: HOSTS file written to /etc/pihole/hosts/custom.list" /var/log/pihole/FTL.log'
+  run bash -c 'grep -c "DEBUG_CONFIG: HOSTS file written to /opt/etc/pihole/hosts/custom.list" /var/log/pihole/FTL.log'
   printf "%s\n" "${lines[@]}"
   [[ ${lines[0]} == "3" ]]
 }
