diff --git a/scripts/js/footer.js b/scripts/js/footer.js
index 63197d35..171ab748 100644
--- a/scripts/js/footer.js
+++ b/scripts/js/footer.js
@@ -479,6 +479,8 @@ function updateVersionInfo() {
     var version = data.version;
     var updateAvailable = false;
     var dockerUpdate = false;
+    var entwareUpdate = false;
+    var packaged = version.docker.local || version.entware.local;
     var isDocker = false;
     $("#versions").empty();
 
@@ -491,6 +493,14 @@ function updateVersionInfo() {
         hash: null,
         url: "https://github.com/pi-hole/docker-pi-hole/releases",
       },
+      {
+        name: "Entware Package",
+        local: version.entware.local,
+        remote: version.entware.remote,
+        branch: version.entware.branch,
+        hash: null,
+        url: "https://github.com/jacklul/entware-pi-hole/releases",
+      },
       {
         name: "Core",
         local: version.core.local.version || "N/A",
@@ -521,7 +531,7 @@ function updateVersionInfo() {
     ];
 
     // Check if we are running in a Docker container
-    if (version.docker.local !== null) {
+    if (version.docker.local !== null || version.entware.local !== null) {
       isDocker = true;
     }
 
@@ -575,6 +585,25 @@ function updateVersionInfo() {
           }
         }
 
+        if (v.name === "Entware Package") {
+          if (v.branch === "pi-hole") { // Display the link to the current release when running the main package
+            localVersion =
+              '<a href="' +
+              v.url +
+              "/" +
+              localVersion +
+              '" rel="noopener" target="_blank">' +
+              localVersion +
+              "</a>";
+          }
+
+          if (versionCompare(v.local.replace(/\D/g, '.'), v.remote.replace(/\D/g, '.')) === -1) {
+            updateComponentAvailable = true;
+            entwareUpdate = true;
+            isDocker = false;
+          }
+        }
+
         // Display update information of individual components only if we are not running in a Docker container
         if ((!isDocker || v.name === "Docker Tag") && updateComponentAvailable) {
           $("#versions").append(
@@ -598,7 +626,11 @@ function updateVersionInfo() {
       $("#update-hint").html(
         'To install updates, <a href="https://github.com/pi-hole/docker-pi-hole#upgrading-persistence-and-customizations" rel="noopener" target="_blank">replace this old container with a fresh upgraded image</a>.'
       );
-    else if (updateAvailable)
+    else if (entwareUpdate)
+      $("#update-hint").html(
+        'To install updates, run <code><a href="https://github.com/jacklul/entware-pi-hole#installation" rel="noopener" target="_blank">opkg update && opkg upgrade pi-hole</a></code>.'
+      );
+    else if (updateAvailable && !packaged)
       $("#update-hint").html(
         'To install updates, run <code><a href="https://docs.pi-hole.net/main/update/" rel="noopener" target="_blank">pihole -up</a></code>.'
       );
